src/contracts/libraries/Merkle.sol

```diff
@@ -123,8 +125,6 @@ library Merkle {
     /**
      @notice this function returns the merkle root of a tree created from a set of leaves using sha256 as its hash function
      @param leaves the leaves of the merkle tree
-
-     @notice requires the leaves.length is a power of 2
```

Comment:
I thought we wanted to *clarify* this wording, not delete it entirely.
The usage of 'require' here was apparently to signal a pre-condition of calling this.
Purely deleting the comment seems unhelpful and undesirable.

Reply:
yeah but as far as the library is concerned, the number of leaves is irrelevant.  So I thought its better to remove it to keep the library as general as possible.   We make the check and comment on it in the BeaconChainProofs, where this condition matters

Reply:
As far as the Merkle library is, it doesn't matter what the length of the leaves is.  It feels like making comments about preconditions seems inappropriate for a general library.  There are comments in the relevant  protocol code

Reply:
I completely disagree. This code is only intended for having an input whose length is a power of 2. It will cut off other inputs improperly.
Noting this behavior in a comment _purely helps protect other people who might use this code_ which was the intention of the original comment. Just removing the comment means making it easier for people to make mistakes in usage.

Reply:
shit for some reason I thought I remembered this thing used zero hashes to pad but it totally doesn't.  Honestly I think a require statement might be the right thing here.

Reply:
the runtime of checking whether a number is a power of two (without witnesses) is logarithmic. i think it's better we expect users of this function to know the exact expected length and make a constant time check before feeding into this function?

Reply:
I'm OK with log-cost checking, seems reasonable to add. An extra check for correctness is likely worth the gas. We can always benchmark it and see the cost.
It's totally infeasible to use this to build a big tree, so number of steps shouldn't be too high, right?